### PR TITLE
Add print,barnstar,delete feature and info of node to ellipsis and add test

### PR DESCRIPTION
**Pull request by CleverFool77**
Fixes https://github.com/publiclab/plots2/issues/6097

### CODE REVIEW

**Comment by CleverFool77** [comment]
Earlier we had popover for all the toggle icons which is okk, but it limits with lots of stuff. As We might have sub-drop-down for barnstars and various other feature. So I'm converting all the data-content in popovers to dropdowns. I wanted to ask that do we have buttons for `Spam`, `Delete, `Raw` to be added in new dropdown ?

**Comment by jywarren** [comment]
Yes, let's add those too!

Do you think you could try to add a system test task that clicks on the new
button so we can see the menu in the screenshots?

**Comment by CleverFool77** [comment]
sure i'll add these. Regarding system test, do you mean clicking on toggle.
Icons and having a screenshot of the menu which is Opened? If this is the
case, then I'll add this test.
Thanks

**Comment by CleverFool77** [comment]
Hi @jywarren 
These are the screenshots - 
### Sidebar in Notes
Screenshot from 2019-08-30 17-19-47

### Sidebar in notes with expanded submenu for users who liked
Screenshot from 2019-08-30 17-19-54

### Sidebar in notes with expanded submenu for barnstar award
Screenshot from 2019-08-30 17-20-00

### Sidebar in Wikis
Screenshot from 2019-08-30 17-19-11

### Sidebar in question
Screenshot from 2019-08-30 17-19-29

**Comment by CleverFool77** [comment]
HI @jywarren I was going for system test, but I had doubt regarding the url. To have screenshot of the sidebar toggle icon dropdowns. We need one url which is either of wiki,notes or questions. So under which url we should be redirected to get the screenshot ?

**Comment by CleverFool77** [comment]
restarting the build.

**Comment by jywarren** [comment]
Yes, you can do:

```ruby
visit node(:about).path
```

in a system test! 👍

**Comment by CleverFool77** [comment]
Thanks Jeff. I'll Update the system test.

File: test/system/Sidebar_test.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+require "application_system_test_case"
+
+class SidebarTest < ApplicationSystemTestCase
```
Comment (CleverFool77) [inline test/system/Sidebar_test.rb]: Hi @jywarren I tried adding sidebar. Is it something like this ?

**Comment by jywarren** [comment]
That looks right! Not sure why other tests are failing, could it be unrelated? I didn't have a moment to look in but I did restart them. 

The only thing you may need is to add an assert_select to check that the dropdown is open and visible before taking the screenshot. That also inserts a short delay to let it appear properly. I believe this is demonstrated in some of the other system tests. Want to give it a try? 

Hopefully the test re-run generates screenshots and we get to see!

**Comment by CleverFool77** [comment]
HI @jywarren I'll add this part of `assert_select`
Thanks !!

**Comment by CleverFool77** [comment]
Hi @jywarren This PR has been updated.

**Comment by CleverFool77** [comment]
Restarting the build.

**Comment by CleverFool77** [comment]
I'll look into these errors.

**Comment by jywarren** [comment]
Try rebasing this maybe? Icarito merged a fix for test timing that may help even after you solve the test issues!

**Comment by CleverFool77** [comment]
It failed earlier :thinking: . I'll try rebasing again.
Thanks !!

**Comment by CleverFool77** [comment]
Hi @jywarren I checked this again. Its not a `failure` but rather its an `error` in function `test blog`.
I migrated to test db and checked the functional testing for `tag_controller` in PR locally.
The error is this : 
```
ERROR["test_tag_blog", #<Minitest::Reporters::Suite:0x000000000cf09a90 @name="TagControllerTest">, 40.89741337799933]
 test_tag_blog#TagControllerTest (40.90s)
ActionView::Template::Error:         ActionView::Template::Error: undefined method `id' for nil:NilClass
            app/views/like/_like.html.erb:37:in `_app_views_like__like_html_erb__2289860276571068026_109093620'
            app/views/tag/blog.html.erb:65:in `block in _app_views_tag_blog_html_erb__3987271905292335004_109261180'
            app/views/tag/blog.html.erb:46:in `each_with_index'
            app/views/tag/blog.html.erb:46:in `_app_views_tag_blog_html_erb__3987271905292335004_109261180'
            test/functional/tag_controller_test.rb:340:in `block in <class:TagControllerTest>'

```

in this function. 
```
  test 'tag blog' do
    get :blog, params: { id: Tag.last.name }
    assert :success
    assert_not_nil :notes
    assert_not_nil :tags
  end
```

Can I have bit idea that what's the connection between blog and `like.html.erb` . I don't seem to get it. :thinking:

**Comment by jywarren** [comment]
I think this was an issue affecting prior PRs that is unrelated. Have you
recently rebased? Because I /believe/ we resolved it but am not 100% sure.
Can you try rebasing if you haven't recently? Thanks!!!

**Comment by CleverFool77** [comment]
Hi @jywarren I'll try it again.
Thanks !!

**Comment by CleverFool77** [comment]
I just rebased it and pushed it again.
Thanks !!

**Comment by CleverFool77** [comment]
I tried again and it failed again. The test is unretaled. 
```
FAIL["test_graph", #<Minitest::Reporters::Suite:0x0000556026b29068 @name="TagSelectionTest">, 0.3324751719999881]
 test_graph#TagSelectionTest (0.33s)
        Expected: 16
          Actual: 20
        test/unit/tag_selection_test.rb:12:in `block in <class:TagSelectionTest>'
```

**Comment by jywarren** [comment]
Wow, this one is really stuck! I'm going to try restarting tests again :-/

**Comment by jywarren** [comment]
Hmm. Still seeing this error!

```
ERROR["test_tag_blog", #<Minitest::Reporters::Suite:0x000056333870b0e0 @name="TagControllerTest">, 76.97021448700002]
 test_tag_blog#TagControllerTest (76.97s)
ActionView::Template::Error:         ActionView::Template::Error: undefined method `id' for nil:NilClass
            app/views/like/_like.html.erb:36:in `_app_views_like__like_html_erb__2249726030689545791_47388994318380'
            app/views/tag/blog.html.erb:65:in `block in _app_views_tag_blog_html_erb__3653190442603664418_47388999884100'
            app/views/tag/blog.html.erb:46:in `each_with_index'
            app/views/tag/blog.html.erb:46:in `_app_views_tag_blog_html_erb__3653190442603664418_47388999884100'
            test/functional/tag_controller_test.rb:337:in `block in <class:TagControllerTest>'
```

File: app/views/like/_like.html.erb [hunk 2]
```diff
@@ -27,70 +27,114 @@
...
-    <input type='hidden' name='nid' value='<%= node.id %>'/>
-    <input type='submit' class='btn btn-primary' value='Give'/> &nbsp; <a _target='blank' href='/wiki/barnstars'>Learn more</a>
-    </form>
+  <div class="dropdowna">
+  <li id="ellipsis_sidebar" class="btn btn-outline-secondary btn-circle btn-sm" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
+      <span class="ml-2 ff fa fa-ellipsis-h" style="font-size: 1em;"></span>
+    </li>
+  <ul class="dropdown-menu">
+    <li class="dropdown-item">
+      <a href="/n/<%= @node.id %>"><i class="fa fa-link"></i></a> <span class="d-none d-xl-inline"><a href="/n/<%= @node.id %>">#<%= @node.id %></a></span>
```
Comment (jywarren) [inline app/views/like/_like.html.erb]: Oh! Aha! You need to be using `node` here as it's a local variable in this template! For all instances of `@node` in this file. That should fix it!

Comment (CleverFool77) [reply app/views/like/_like.html.erb]: Ohhh!!! I'll fix it.
Thanks Jeff

**Comment by CleverFool77** [comment]
Hi @jywarren I wanted to ask this.
```
ERROR["test_viewing_the_sidebar", #<Minitest::Reporters::Suite:0x0000564fe316f650 @name="SidebarTest">, 4.181052211999997]
 test_viewing_the_sidebar#SidebarTest (4.18s)
NoMethodError:         NoMethodError: undefined method `node' for #<SidebarTest:0x0000564fe3bc8138>
        Did you mean?  nodes
            test/system/Sidebar_test.rb:7:in `block in <class:SidebarTest>'
```
Is `node` and `@node` going to effect the tests too ? What does this mean of undefined method `node` in error. Is it this `visit node(:about).path` in sidebar test ?

**Comment by jywarren** [comment]
Hmm. Can you find the line causing the error? It may need an "if node && "
or something, or it might need the node local to be defaulted to @node,
like node |= @node

But it depends on the context!

**Comment by jywarren** [comment]
I think this is now solved! Thanks!!

**Pull request closed without merging**